boot/bootutil/src/loader.c

```diff
@@ -3230,7 +3253,7 @@ context_boot_go(struct boot_loader_state *state, struct boot_rsp *rsp)
 out:
     close_all_flash_areas(state);
 
-    if (fih_eq(fih_rc, FIH_SUCCESS)) {
+    if (rc != 0) {
```

Comment:
FIH_SET in the next line? then might the individual calls in front of the goto statements could be removed

Reply:
To obtain the best security, you have to switch as soon as possible from weak `rc` return values to more robust `fih_rc`. affectations before got were added to improve the code security and therefore cannot be removed without introducing vulnerabilities.